### PR TITLE
Prevent EnC error when changing partial methods with lambdas

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/60804

### CODE REVIEW

**Comment by davidwengier** [comment]
@tmat PTAL

**Review by cston — APPROVED** [review]

**Review by tmat — APPROVED** [review]
:shipit:

**Comment by tmat** [comment]
Nice!

File: src/Compilers/Core/Portable/Emit/EditAndContinue/SymbolChanges.cs [hunk 1]
```diff
@@ -191,6 +191,15 @@ public SymbolChange GetChange(IDefinition def)
 
         private SymbolChange GetChange(ISymbol symbol)
         {
+            // In CalculateChanges we always store definitions for partial methods, so we have to
+            // make sure we do the same thing here when we try to retreive a change, as the compiler
```
[the remark this answers is not in the document]
Comment (davidwengier) [reply src/Compilers/Core/Portable/Emit/EditAndContinue/SymbolChanges.cs:195]: I should write a keylogger that detects when I write any word with an i and an e next to each other, and have it pop up a message box saying "you probably got that wrong"

**Pull request merged**